src/error.rs

```diff
@@ -213,6 +213,27 @@ impl Error {
         }
     }
 
+    /// Returns true, if a network error occured.
+    #[inline]
+    pub fn is_network_error(&self) -> bool {
+        match self.inner.kind {
+            Kind::Hyper(ref e) => {
+                if e.is_canceled() {
```

Comment:
You could collapse these into a single `if e.is_canceled() || e.is_connected() || etc`.

Reply:
done